### PR TITLE
Add /nodes pages

### PR DESCRIPTION
**Pull request by VallariAg**
New pages:
- [x] [/nodes](https://pulpito.ceph.com/nodes/)
Screenshot 2023-08-10 at 4 56 47 PM

- [x] [/nodes/:name](https://pulpito.ceph.com/nodes/smithi001.front.sepia.ceph.com/) 
Screenshot 2023-08-11 at 2 54 51 PM

- [x] [node lock stats](https://pulpito.ceph.com/stats/nodes/locks/)
Screenshot 2023-08-14 at 1 11 03 PM

- [x] [node jobs stats](https://pulpito.ceph.com/stats/nodes/jobs/)
Screenshot 2023-08-14 at 7 53 29 PM

### CODE REVIEW

File: src/lib/paddles.ts [hunk 1]
```diff
@@ -94,6 +94,22 @@ function useMachineTypes() {
   return useQuery(["machine_types", { url }]);
 }
 
+function useNodes(params: GetURLParams): UseQueryResult<Node[]> {
+  const params_ = JSON.parse(JSON.stringify(params || {}));
+
+  const queryString = new URLSearchParams(params_).toString();
+  let uri = "nodes" + (queryString? `?${queryString}` : '');
```
Comment (zmc) [inline src/lib/paddles.ts:101]:
```suggestion
  let uri = "nodes/" + (queryString? `?${queryString}` : '');
```
`/nodes/` isn't loading on the PR preview, and I see '302 Found' coming back from paddles. Maybe this'll fix that.

**Review by zmc — CHANGES_REQUESTED** [review]
This looks great, thanks for picking it up!

I found one bug:
* Lock stats don't match https://pulpito.ceph.com/stats/nodes/locks/ (in fact all values are 1)

I see a few minor UI tweaks to make:
* The list should be sorted by the name column by default
* Checkboxes for up and locked are white in dark mode; they should be black. Probably via `theme.text` instead of `theme.palette`
* The description field should be a lot longer

Some change ideas I'm not 100% on:
* Is the machine type column worth the space it takes up, given our naming conventions?
* Is the arch column worth the space? All the non-x86_64s are unused or retired.
* Should we default to showing only smithi machines with the drawer links? Other types are irrelevant currently.
* Should we extend `FilterMenu` to be able to use a [`Select`](https://mui.com/material-ui/react-select/)? (Maybe not an item for this PR)

**Comment by VallariAg** [comment]
@zmc I have made all the changes you mentioned. I found few more bugs along the way and fixed them as well. 
Sorry this has turned into a massive PR!

> Should we extend FilterMenu to be able to use a [Select](https://mui.com/material-ui/react-select/)? (Maybe not an item for this PR)

From UI perspective, the `Autocomplete` component looks pretty similar to the `Select`. Are there any additional benefits of using `Select`?

**Review by kamoltat — APPROVED** [review]
Looks good to me, I guess it's pretty much ready once you fix some of the bugs you said you identified

**Review by zmc — APPROVED** [review]
Awesome! Thanks for addressing all of those things.

Re: the Select comment, the only real difference would be not allowing keyboard input, making the UI a tiny bit more simple. It looks like [`TextField` can become a `Select` via a prop](https://mui.com/material-ui/react-text-field/#select) so if we want that, it'll be trivial. But it's not at all important in this PR.

**Pull request merged**